python/src/scipp/table_html/formatting_html.py

```diff
@@ -292,53 +287,46 @@ def _format_common(is_index):
-                     attrs_ul, disabled, data_repr, is_index, has_attrs):
+def summarize_variable(name,
+                       var,
+                       is_index=False,
+                       has_attrs=False,
+                       bin_edges=None,
+                       add_dim_size=False):
     """
     Formats the variable data into the format expected when displaying
-    as a standalone Variable. This happens when a single Variable or
-    DataArray is displayed.
+    as a standalone variable (when a single variable or data array is
+    displayed) or as part of a dataset.
     """
-    cssclass_idx, attrs_id, attrs_icon, data_id, data_icon = _format_common(
-        is_index)
+    dims_str = escape(f"({_make_dim_str(var, bin_edges, add_dim_size)})")
+    unit = '' if var.unit == sc.units.dimensionless else repr(var.unit)
 
-    html = [
-        f"<div class='sc-var-name'><span{cssclass_idx}>{dims_str}</span>"
-        "</div>",
-        f"<div class='xr-var-dtype'>{escape(dtype)}</div>",
-        f"<div class='xr-var-unit'>{escape(unit)}</div>",
-        f"<div class='xr-value-preview xr-preview'><span>{preview}</span>",
-        "{}</div>".format(f'<span>{variances_preview}</span>'
-                          if variances_preview is not None else ''),
-        f"<input id='{attrs_id}' class='xr-var-attrs-in' ",
-        f"type='checkbox' {disabled}>",
-        f"<label for='{attrs_id}' "
-        f"class='{'' if has_attrs else 'xr-hide-icon'}'"
-        " title='Show/Hide attributes'>",
-        f"{attrs_icon}</label>",
-        f"<input id='{data_id}' class='xr-var-data-in' type='checkbox'>",
-        f"<label for='{data_id}' title='Show/Hide data repr'>",
-        f"{data_icon}</label>",
-        f"<div class='xr-var-attrs'>{attrs_ul}</div>" if attrs_ul else "",
-        f"<pre class='xr-var-data'>{data_repr}</pre>",
-    ]
-    return "".join(html)
+    disabled, attrs_ul = _make_inline_attributes(var, has_attrs)
 
+    preview = inline_variable_repr(var)
+    data_repr = f"Values:<br>{short_data_repr_html(var)}"
+    variances_preview = None
+    if var.variances is not None:
+        variances_preview = inline_variable_repr(var, has_variances=True)
+        data_repr += f"<br><br>Variances:<br>\
+                       <div>{short_data_repr_html(var)}</div>"
 
-def _dataset_format(name, dims_str, dtype, unit, preview, variances_preview,
-                    attrs_ul, disabled, data_repr, is_index, has_attrs):
-    """
-    Formats the variable data into the format expected when displaying
-    as part of a dataset.
-    """
     cssclass_idx, attrs_id, attrs_icon, data_id, data_icon = _format_common(
         is_index)
 
-    html = [
-        f"<div class='xr-var-name'><span{cssclass_idx}>{escape(name)}</span>"
-        "</div>",
-        f"<div class='xr-var-dims'>{escape(dims_str)}</div>",
-        f"<div class='xr-var-dtype'>{escape(dtype)}</div>",
+    if name is None:
```

Comment:
Could we change this to
`if name`
then swap the branches. As it will handle an empty name (if that's possible) the same as one that doesn't exist

Reply:
> As it will handle an empty name (if that's possible) the same as one that doesn't exist

This is why we *are* checking for `None`: Empty names *are* possible and must be handled like a non-empty name. `None` is different. I believe this is used here to distinguish between summarizing a variable and summarizing the data variable of a data array.

Reply:
Do we explicitly handle empty names the same as populated names then throughout the codebase?

Basically I see
`name:`
as potentially confusing, but equally I understand why it's valid if we handle it consistently, hence I raise it?

Reply:
By default a `DataArray` has no name. We don't do anything to names, so they can be arbitrary, including empty. No magic strings, unlike the Mantid ADS!